crates/types/src/locality/location_scope.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2023 - 2025 Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+// [[NodeLocationScope]] specifies the location of a node in the cluster. The location
+// is expressed by a set of hierarchical scopes. Restate assumes the cluster topology
+// to be a tree-like structure: two nodes sharing a lower-level scope must
+// be within the same higher level scope.
```

Comment:
Does this mean that scopes must have unique names? So there can only be one `a1` AZ and not `us-west.a1` and `us-east.a1`?

Reply:
Not really. I'll remove part of this paragraph since it adds confusion.